enhancements/update/targeted-update-edge-blocking.md

```diff
@@ -0,0 +1,247 @@
+
+## Proposal
+
+### Enhanced graph-data schema for blocking edges
+
+[The blocked-edges schema][block-edges] will be extended with the following new properties:
+
+* `clusters` (optional, [object][json-object]), defining the subset of affected clusters.
+    If any `clusters` property matches a given cluster, the edge should be blocked for that cluster.
+    * `promql` (optional, [object][json-string]), with a [PromQL][] query describing affected clusters.
+        This query will be submitted to a configurable Prometheus service and should return a set of matching records with `_id` labels.
+        Clusters whose [submitted `id` query parameter][cincinnati-for-openshift-request] is in the set of returned IDs are allowed if the value of the matching record is 1 and blocked if the value of the matching record is 0.
+        Clusters whose submitted `id` is not in the result set, or which provide no `id` parameter, are also blocked.
+        Blocking too many edges is better than blocking too few, because you can recover from the former, but possibly brick clusters on the latter.
+
+        The result of the query may be cached by the update service, so queries which generate unstable result sets should be avoided.
+        Clusters whose submitted `id` is not in the result set may be considered cache misses and, if a cache refresh still fails to include them, may be cached as `blocked`.
+
+[The schema version][graph-data-schema-version] would also be bumped to 1.1.0, because this is a backwards-compatible change.
+Consumers who only understand graph-data schema 1.0.0 would ignore the `clusters` property and block the edge for all clusters.
+While blocking the edge for all clusters is more aggressive than the graph-data maintainers intended, it is failing on the side of safety.
+Blocking these edges for all clusters is also not as aggressive as complaining about an unrecognized 2.0.0 version and failing to serve the entire graph.
+
+### Update service support for the enhanced schema
+
+The following recommendations are geared towards the [openshift/cincinnati][cincinnati].
+Maintainers of other update service implementations may or may not be able to apply them to their own implementation.
+
+The graph-builder's graph-data scraper should learn about [the new 1.1.0 schema](#enhanced-graph-data-schema-for-blocking-edges), and record any `clusters` properties in edge metadata.
+
+When a request is received, the submitted [`channel` query parameter][cincinnati-for-openshift-request] limits the set of remaining edges.
+If any of the remaining edges have `clusters.promql` entries, a new, targeted-edge-blocking policy engine plugin will exclude the edge if the [`id` query parameter][cincinnati-for-openshift-request] is a member of the query's resulting ID set.
+If the request does not set the `id` parameter, the plugin should block all conditional edges, and does not need to check a cache or make PromQL queries.
```

Comment:
>  I am not sure what is conditional edges mean.

Edges which have one of these conditions blocks.  No need to block edges if you know that all clusters should see that edge.

Reply:
Defined the "conditional edges" terminology:

```console
$ git --no-pager diff -U1 600f8b4..0d8c588
diff --git a/enhancements/update/targeted-update-edge-blocking.md b/enhancements/update/targeted-update-edge-blocking.md
index cfb3b89..723d3e4 100644
--- a/enhancements/update/targeted-update-edge-blocking.md
+++ b/enhancements/update/targeted-update-edge-blocking.md
@@ -87,2 +87,3 @@ Maintainers of other update service implementations may or may not be able to ap
 The graph-builder's graph-data scraper should learn about [the new 1.1.0 schema](#enhanced-graph-data-schema-for-blocking-edges), and record any `clusters` properties in edge metadata.
+Blocked edges with `clusters` containing a conditional rule like `promql` are "conditional edges" in the following discussion, because the presence of an edge in the recommended update graph returned to a client depends on whether the rule conditions apply to the client cluster or not.
 
```